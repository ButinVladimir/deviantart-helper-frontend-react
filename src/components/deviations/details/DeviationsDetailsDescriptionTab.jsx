import React from 'react';
import PropTypes from 'prop-types';
import Section from 'react-bulma-components/lib/components/section';
import Container from 'react-bulma-components/lib/components/container';
import Content from 'react-bulma-components/lib/components/content';
import Box from 'react-bulma-components/lib/components/box';
import Media from 'react-bulma-components/lib/components/media';
import DeviationThumbnail from '../shared/DeviationThumbnail';
import StatisticsLevel from '../../shared/StatisticsLevel';

/* eslint-disable react/no-danger */

export default function DeviationsDetailsDescriptionTab({
  title,
  thumbnail,
  url,
  publishedTime,
  description,
  views,
  favourites,
  comments,
  downloads,
}) {
  return (
    <>
      <Section>
        <Container>
          <Media>
            <DeviationThumbnail
              title={title}
              thumbnail={thumbnail}
              url={url}
              publishedTime={publishedTime}
            />
            <Media.Item>
              <Content dangerouslySetInnerHTML={{ __html: description }} />
            </Media.Item>
          </Media>
        </Container>
      </Section>

      <Section>
        <Container>
          <Box>
            <StatisticsLevel
              titleValuePairs={[['Views', views], ['Favourites', favourites], ['Comments', comments], ['Downloads', downloads]]}
            />
          </Box>
        </Container>
      </Section>
    </>
  );
}

DeviationsDetailsDescriptionTab.propTypes = {
  ...DeviationThumbnail.PropTypes,
  url: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  views: PropTypes.number.isRequired,
  favourites: PropTypes.number.isRequired,
  comments: PropTypes.number.isRequired,
  downloads: PropTypes.number.isRequired,
};

/* eslint-enable no-danger */
