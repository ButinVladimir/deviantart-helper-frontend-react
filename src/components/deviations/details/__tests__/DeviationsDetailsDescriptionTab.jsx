import React from 'react';
import { shallow } from 'enzyme';
import DeviationsDetailsDescriptionTab from '../DeviationsDetailsDescriptionTab';

describe('DeviationsDetailsDescriptionTab', () => {
  it('can be rendered correctly', () => {
    const wrapper = shallow(
      <DeviationsDetailsDescriptionTab
        title="Deviation title"
        thumbnail={{
          src: '/path/to/thumbnail',
          width: 1,
          height: 2,
        }}
        url="https://path/to/deviation"
        publishedTime={4}
        description="Description"
        nsfw
        views={1}
        favourites={2}
        comments={3}
        downloads={4}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
