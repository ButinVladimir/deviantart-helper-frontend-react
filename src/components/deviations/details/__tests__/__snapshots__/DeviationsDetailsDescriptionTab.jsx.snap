// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviationsDetailsDescriptionTab can be rendered correctly 1`] = `
<Fragment>
  <Section
    className=""
    renderAs="section"
    size={null}
    style={Object {}}
  >
    <Container
      breakpoint={null}
      fluid={false}
      renderAs="div"
      style={Object {}}
    >
      <Media
        className=""
        renderAs="article"
        style={Object {}}
      >
        <DeviationThumbnail
          nsfw={true}
          publishedTime={4}
          thumbnail={
            Object {
              "height": 2,
              "src": "/path/to/thumbnail",
              "width": 1,
            }
          }
          title="Deviation title"
          url="https://path/to/deviation"
        />
        <MediaItem
          className=""
          position="center"
          renderAs="div"
          style={Object {}}
        >
          <Content
            dangerouslySetInnerHTML={
              Object {
                "__html": "Description",
              }
            }
            renderAs="div"
            size={null}
            style={Object {}}
          />
        </MediaItem>
      </Media>
    </Container>
  </Section>
  <Section
    className=""
    renderAs="section"
    size={null}
    style={Object {}}
  >
    <Container
      breakpoint={null}
      fluid={false}
      renderAs="div"
      style={Object {}}
    >
      <Box
        className=""
        renderAs="div"
        style={Object {}}
      >
        <StatisticsLevel
          titleValuePairs={
            Array [
              Array [
                "Views",
                1,
              ],
              Array [
                "Favourites",
                2,
              ],
              Array [
                "Comments",
                3,
              ],
              Array [
                "Downloads",
                4,
              ],
            ]
          }
        />
      </Box>
    </Container>
  </Section>
</Fragment>
`;
